pkg/http/api_test.go

```diff
@@ -2284,12 +2284,14 @@ func TestV1Alpha1ConfigurationUnsubscribe(t *testing.T) {
 	t.Run("subscribe and unsubscribe configurations", func(t *testing.T) {
 		apiPath1 := fmt.Sprintf("v1.0-alpha1/configuration/%s/subscribe", storeName)
 		resp1 := fakeServer.DoRequest("GET", apiPath1, nil, nil)
-		assert.Equal(t, 200, resp1.StatusCode, "subscribe configuration store, should return 200")
+		assert.Equal(t, 500, resp1.StatusCode, "subscribe configuration store, should return 500 when app channel is empty")
```

Comment:
Do we have test scenarios for the happy path still?

Reply:
We are mocking a fake http server  with the endpoints of interest - 

https://github.com/akhilac1/dapr/blob/8130e7ffca658c6ff4984722ad307ae80bd7ed82/pkg/http/api_test.go#L2174

Happy path would need initialization of the api instance with NewAPI func. So we cannot test happy path subscribe in unit tests.

Reply:
Why did we have a test checking for http 200?

Reply:
Earlier subscribe call was always passing as long as the name of the configuration store was accurate. There was no check to see if the notifications can be actually consumed. In scenarios where no application was connected and appchannel was nil - code panicked and this was handled in the defer building block. We could do 2 things - 
1. check if app channel is nil on receiving notification from configuration store and silently drop the notification
2. block subscription if the notifications cannot be consumed due to app not being ready or app not available to consume notifications. 
This implementation ensures that there is a consumer for notifications and disallows subscription otherwise using the appchannel check for http subscribe API call.